cpp/tests/fixed_point/fixed_point_tests.cpp

```diff
@@ -45,67 +46,71 @@ TYPED_TEST(FixedPointTestAllReps, SimpleDecimalXXConstruction)
+  auto num0 = cudf::convert_floating_to_fixed<decimalXX>(-1.234567, scale_type(0));
+  auto num1 = cudf::convert_floating_to_fixed<decimalXX>(-1.234567, scale_type(-1));
+  auto num2 = cudf::convert_floating_to_fixed<decimalXX>(-1.234567, scale_type(-2));
+  auto num3 = cudf::convert_floating_to_fixed<decimalXX>(-1.234567, scale_type(-3));
+  auto num4 = cudf::convert_floating_to_fixed<decimalXX>(-1.234567, scale_type(-4));
+  auto num5 = cudf::convert_floating_to_fixed<decimalXX>(-1.234567, scale_type(-5));
+  auto num6 = cudf::convert_floating_to_fixed<decimalXX>(-1.234567, scale_type(-6));
+
+  EXPECT_EQ(-1, cudf::convert_fixed_to_floating<double>(num0));
+  EXPECT_EQ(-1.2, cudf::convert_fixed_to_floating<double>(num1));
+  EXPECT_EQ(-1.23, cudf::convert_fixed_to_floating<double>(num2));
+  EXPECT_EQ(-1.234, cudf::convert_fixed_to_floating<double>(num3));
+  EXPECT_EQ(-1.2345, cudf::convert_fixed_to_floating<double>(num4));
+  EXPECT_EQ(-1.23456, cudf::convert_fixed_to_floating<double>(num5));
+  EXPECT_EQ(-1.234567, cudf::convert_fixed_to_floating<double>(num6));
 }
 
 TYPED_TEST(FixedPointTestAllReps, PaddedDecimalXXConstruction)
 {
   using decimalXX = fixed_point<TypeParam, Radix::BASE_10>;
 
-  decimalXX a{1.1, scale_type{-1}};
-  decimalXX b{1.01, scale_type{-2}};
-  decimalXX c{1.001, scale_type{-3}};
-  decimalXX d{1.0001, scale_type{-4}};
-  decimalXX e{1.00001, scale_type{-5}};
-  decimalXX f{1.000001, scale_type{-6}};
-
-  decimalXX x{1.000123, scale_type{-8}};
-  decimalXX y{0.000123, scale_type{-8}};
-
-  EXPECT_EQ(1.1, static_cast<double>(a));
-  EXPECT_EQ(1.01, static_cast<double>(b));
-  EXPECT_EQ(1, static_cast<double>(c));  // intentional (inherited problem from floating point)
-  EXPECT_EQ(1.0001, static_cast<double>(d));
-  EXPECT_EQ(1.00001, static_cast<double>(e));
-  EXPECT_EQ(1, static_cast<double>(f));  // intentional (inherited problem from floating point)
-
-  EXPECT_TRUE(1.000123 - static_cast<double>(x) < std::numeric_limits<double>::epsilon());
-  EXPECT_EQ(0.000123, static_cast<double>(y));
+  auto a = cudf::convert_floating_to_fixed<decimalXX>(1.1, scale_type(-1));
+  auto b = cudf::convert_floating_to_fixed<decimalXX>(1.01, scale_type(-2));
+  auto c = cudf::convert_floating_to_fixed<decimalXX>(1.001, scale_type(-3));
+  auto d = cudf::convert_floating_to_fixed<decimalXX>(1.0001, scale_type(-4));
+  auto e = cudf::convert_floating_to_fixed<decimalXX>(1.00001, scale_type(-5));
+  auto f = cudf::convert_floating_to_fixed<decimalXX>(1.000001, scale_type(-6));
+  auto x = cudf::convert_floating_to_fixed<decimalXX>(1.000123, scale_type(-8));
+  auto y = cudf::convert_floating_to_fixed<decimalXX>(0.000123, scale_type(-8));
+
+  EXPECT_EQ(1.1, cudf::convert_fixed_to_floating<double>(a));
+  EXPECT_EQ(1.01, cudf::convert_fixed_to_floating<double>(b));
+  EXPECT_EQ(1,
+            cudf::convert_fixed_to_floating<double>(
+              c));  // intentional (inherited problem from floating point)
```

Comment:
Can you help me understand this comment?

Reply:
This sort of thing happens when you type something like 1.001, but the closest floating point value to that is 1.0009999, and the truncation to 3 decimal places truncates that to 3 decimal places instead of rounding (this choice is by design).  This problem is fixed in my upcoming conversion code, but that'll be in a future PR.